logos/benchmarks/run_bench.sh

```
@@ -0,0 +1,109 @@
+       "num_samples=${NUM_SAMPLES:-ALL}) ..."
+  prepare_args=(--split "$GSM8K_SPLIT" --rps "$GSM8K_RPS")
+  [[ -n "$NUM_SAMPLES" ]] && prepare_args+=(--num-samples "$NUM_SAMPLES")
+  "$PYTHON" -u prepare_benchmark.py "${prepare_args[@]}"
+fi
+
+# ── Step 2: run the benchmark ─────────────────────────────────────────────────
+bench_args=(
+  --run-all-scenarios
+  --logos-url "$LOGOS_URL"
+  --workload "$WORKLOAD"
+  --gpu-host $GPU_HOSTS
```

Shellcheck (0.11.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)